### DISCUSSION TITLE
Issue with pushViewController inside native UIViewControllers

### DISCUSSION THREAD

**Discussion by mohsinulhaq** [software-mansion/react-native-screens#866] [Q&A] [answered]
I have RNScreen native stack setup with a real native screen mounted from Swift. When I try to call ` navigationController.pushViewController` with `animated: true` to navigate to another controller, I get the following exception:

```
(lldb) po navigationController?.pushViewController(FriendsViewController(), animated: true)
2021-03-24 22:09:23.714295+0530 CureFit[97023:7435834] -[UIView stackAnimation]: unrecognized selector sent to instance 0x7fc7d3f8b790
error: Execution was interrupted, reason: internal ObjC exception
```
`stackAnimation` being a RNScreen construct. With animation: false all works fine. But I would like the stack animation to work even from outside React Native. How can I achieve that now?

**Comment by WoLewicki** [comment]
I think you have to resolve the internals by yourself unfortunately. The `stackAnimation` is not needed for `tab` navigator for sure, so you probably use this prop somewhere in your code, where it is not meant to be used. I cannot say much more. The `stackAnimation` resolves to different native transition styles (see: https://github.com/software-mansion/react-native-screens/blob/master/ios/RNSScreen.m#L133).

**Reply by mohsinulhaq** [reply, marked as answer]
@WoLewicki 
I did debug the cause of this since posting this discussion. I am definitely not using the prop `stackAnimation` myself.
The error occurs at this [line](https://github.com/software-mansion/react-native-screens/blob/master/ios/RNSScreenStack.m#L134).
And it's happening because the View I'm trying to push is not a `RNScreenView` instance but a simple `UIView` instance, which doesn't contain the `stackAnimation` property. Hence resulting in the above error.

**Reply by WoLewicki** [reply]
Ok, nice that you managed to do it!

**Reply by mohsinulhaq** [reply]
@WoLewicki yes, I did manage till here. Just that now I'm having issues with consolidating `UIViewControlller` with `RNSScreen` and `UINavigationController` with `RNSStack`.
